lib/util/LocationStatusManager.js

```diff
@@ -252,14 +253,31 @@ class LocationStatusManager {
             { upsert: false })
             .then(() => cb())
             .catch(err => {
-                if (err) {
-                    this._logger.error('Could not update location service status in MongoDB', {
-                        method: 'LocationStatusManager._updateServiceStatusForLocation',
-                        error: err.message,
-                    });
-                    return cb(err);
-                }
-                return cb();
+                this._logger.error('Could not update location service status in MongoDB', {
+                    method: 'LocationStatusManager._updateServiceStatusForLocation',
+                    error: err.message,
+                });
+                
+                const services = Object.keys(previousState);
+                services.forEach(service => {
+                    if (previousState[service] === null) {
+                        return;
+                    }                        
+                    if (previousState[service].paused) {
+                        this._locationStatusStore[location].pauseLocation(service);
+                    } else {
+                        this._locationStatusStore[location].resumeLocation(service, null);
+                    }                            
+                    if (previousState[service].scheduledResume) {
+                        this._locationStatusStore[location].setServiceResumeSchedule(
+                            service, 
+                            new Date(previousState[service].scheduledResume)
+                        );
+                    } else {
+                        this._locationStatusStore[location].setServiceResumeSchedule(service, null);
+                    }  
```

Comment:
An easier approach would be to initialize the `LocationStatus` class with the value of the older one.

```
// example
const previousState = this._locationStatusStore[location].getValue();
this._locationStatusStore[location] = new LocationStatus(this._supportedServices, previousState);
```